src/cljs/main/broadfcui/common/method/config_io.cljs

```diff
@@ -0,0 +1,139 @@
+                                                                (merge table-style/table-cell-plank-middle
+                                                                       table-style/table-cell-optional)
+                                                                table-style/table-cell-plank-middle))} variable])}
+                               {:header "Type" :initial-width 120
+                                :sort-by :type
+                                :as-text (fn [{:keys [type optional?]}]
+                                           (str type (when optional? " (optional)")))
+                                :render
+                                (fn [{:keys [type optional?]}]
+                                  [:div {:style (clip (if optional?
+                                                        (merge table-style/table-cell-plank-right
+                                                               table-style/table-cell-optional)
+                                                        table-style/table-cell-plank-right))}
+                                   type])}]
+                              (when values
+                                [{:header "Attribute" :initial-width 200
+                                  :as-text :value
+                                  :sort-by :text
+                                  :render
+                                  (fn [{:keys [name value optional?]}]
+                                    (let [value (when-not (string/blank? value) value)]
+                                      [:div {:style (clip table-style/default-cell-left)}
+                                       (if editing?
+                                         ;; (ab)using TagAutocomplete instead of Typeahead because it
+                                         ;; plays nicer with tables
+                                         [comps/TagAutocomplete
+                                          {:show-counts? false :allow-clear? true
+                                           :minimum-input-length 0
+                                           :tags [value] :maximum-selection-length 1
+                                           :language {:maximumSelected (fn [] "")}
+                                           ;; `value` ensures that custom selections are initially selected when going to edit
+                                           ;; `distinct` because having multiple copies of the same screws things up
+                                           :data (distinct (if value (conj data value) data))
+                                           :placeholder (if optional? "Optional" "Select or enter")
+                                           :on-change (fn [value]
+                                                        (swap! locals update io-key assoc (keyword name)
+                                                               (if (empty? value) "" (first value))))}]
```

Comment:
1. enter a value for an attribute, like "workspace.foo", then hit enter. "workspace.foo" is now my value.
2. click after "foo". Note that you get a cursor and are allowed to keep typing.
3. add "bar" so that you now have "workspace.foobar". Try to save this value - it doesn't work, and you revert back to "workspace.foo".

I think we should either allow users to edit the attribute, or require them to delete/re-add - but in the latter case we shouldn't give them a live cursor making them think they can edit it.

Reply:
Honestly, the best solution to this is to roll our own typeahead for situations like this. We're already massively hacking they tag feature of this one. I don't think we have a solution to this problem in particular.

Reply:
(it thinks you're trying to add a second tag, rather than edit the one in place)